0016-pre-shared-key/0016-pre-shared-key.md

```diff
@@ -0,0 +1,77 @@
+# Pre-Shared Key
+
+The Pre-Shared Key (PSK) protocol is an end-to-end transport protocol, used by the sender and receiver of an ILP payment to decide on a condition and fulfillment for a payment. The protocol also encrypts any additional data sent along with the payment. The encryption algorithm used is AES-256-CTR.
```

Comment:
You'll want to be *really* careful about using an unauthenticated encryption mode like CTR. I haven't read through how you're doing the key derivation with respect to the AES and HMAC keys, but unless you can guarantee they're always derived together and only together, you're opening yourself up to chosen ciphertext attacks.

Using an AEAD mode (e.g. AES-256-GCM) is the simplest way to avoid any problems regarding IND-CCA security.